Finished documenting IsAny-x shards and Input shards

ShahFragcolor opened the pull request:
- Added input help, output help and help texts for Input shards
- Added input help, output help and help texts for IsAny, IsAll , IsAllEqual etc.shards
- Added test script for IsAny-x shards
- changed IgnoreConsumed parameter on some Input shard to SkipConsumed for better clarity
- Removed None types from Consume and SkipConsumed parameters and ensured they had the appropriate default values


<!--
Before submitting your PR, please review the following checklist:

- [ ] **DO NOT** submit a PR without having discussed the change first in a related issue. If none exist, create one first.
- [ ] **CONSIDER** adding a unit test if your PR resolves an issue.
- [ ] **DO** keep pull requests small so they can be easily reviewed.
- [ ] **DO** make sure all unit tests pass.
- [ ] **DO** make sure not to introduce any new compiler warnings.
- [ ] **AVOID** breaking the continuous integration build.
- [ ] **AVOID** making significant changes to the overall architecture.
-->

ShahFragcolor committed:
Finished documenting IsAny-x shards and Input shards

- Added input help, output help and help texts for Input shards
- Added input help, output help and help texts for IsAny, IsAll , IsAllEqual etc.shards
- Added test script for IsAny-x shards
- changed IgnoreConsumed parameter on some Input shard to SkipConsumed for better clarity
- Removed None types from Consume and SkipConsumed parameters and ensured they had the appropriate default values

sinkingsugar commented on shards/core/runtime.hpp:
```diff
@@ -238,6 +238,8 @@ struct DefaultHelpText {
 
   static inline const SHOptionalString InputHelpAnyType = SHCCSTR("Input of any type is accepted.");
 
+  static inline const SHOptionalString InputHelpAnyButType = SHCCSTR("Input of any type is accepted. However, when comparing different types(e.g., boolean with integer), the shard will not throw an error, but will sometimes return unexpected results.");
```
not sure on `unexpected`, sounds a bit "bad" ? haha
They are more "implementation detail" results, not sure how to define it.

sinkingsugar replied:
lexicographical results

ShahFragcolor committed:
Edited InputHelpAnyButType and BaseOpsBin parameter help text

- edited the InputHelpAnyButType to be more accurate
- edited BaseOpsBin parameter help text for the IsAny-x shards so that the help text for the Value parameter includes information about evaluating nested variables.
